### ISSUE TITLE
ChunkPurge v1.7

### ISSUE THREAD

**Issue by groveham** [stevestech/ChunkPurge#5]
Just wondering if your planning on making a Minecraft 1.7 version of this mod? Seems it would be very useful for my server, and I'm sure many others!

**Comment by stevestech** [comment]
Well this mod was experimental, on my TPPI server I found that there were always some serious lagspikes when the chunk unloading ran, despite my strategy to only unload isolated groups of chunks. These lagspikes also occurred for me when Cauldron's chunkGC method was enabled. I believe Cauldron now sets chunkGC to false by default for this reason.

I think one potential issue is a large group of mods that do not check if a chunk is currently loaded before accessing / editing the terrain. This causes chunks to be quickly reloaded after ChunkPurge runs despite no players or chunkloaders being present, and is most likely the cause of the lagspikes I was experiencing. Cauldron provides an option to prevent chunks from being loaded this way, by returning a null chunk to the mod that requested it, however this just causes a whole lot of null pointer exceptions as the mods do not perform null checks.

That being said, clienthax did update this code to 1.7.10 so it should just be a matter of cloning this repository and building the code. I can do this tonight and upload the jar if you need.

**Comment by Android25** [comment]
I compiled clienthax's code but the mod didn't work with Infitech 2 (which is what grove is looking for). There was a problem when the client proxy injected into the main jar. I forked the project and modified the client proxy, pretty much nulifying it since it's a server side mod.
I compiled it with the forge version for Infitech 2 3.1.14, which may have been the problem with proxy file, as I'm not using cauldron.
Compiling it with the modified clientproxy worked and the mod seemed to work perfectly on a very very small CSP world. I have yet to see how it will effect tps on a much larger forge server.
Have you tried the mod on a stock forge server? I don't know if Cauldron uses forge's chunk handling code of if they modified it to work with plugins.

**Comment by stevestech** [comment]
@Android25 Yeah the mod was made in order to provide the ChunkGC feature from Cauldron for stock Forge servers, as I went through a phase during 1.6 where I wanted to use stock Forge.

I notice that the 1.7.10 source doesn't include allowRemoteVersions = "*", which will be needed in the @Mod annotation in order for it to work as a server-side mod. I'll change this tonight.
